src/main/java/org/apache/commons/lang3/builder/DiffBuilder.java

```diff
@@ -842,6 +842,10 @@ public DiffBuilder append(final String fieldName, final Object lhs,
         }
 
         // Not array type
+        if (left != null ? left.equals(right) : right.equals(left)) {
```

Comment:
:flushed:  Just added a [patch](https://github.com/j--baker/commons-lang/commit/f1a3e786617e50680589684218dc29ad7650c685) to fix this.  This was supposed to be comparing the fields (lhs, rhs), not the containers (left, right).  The containers are checked in the constructor, but this method is concerned with the fields (after getting past the objectsTriviallyEqual test).  Sorry!